ddl/placement_policy.go

```diff
@@ -232,3 +241,53 @@ func onAlterPlacementPolicy(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	job.FinishDBJob(model.JobStateDone, model.StatePublic, ver, nil)
 	return ver, nil
 }
+
+func checkPlacementPolicyNotInUse(d *ddlCtx, t *meta.Meta, policy *model.PolicyInfo) error {
+	currVer, err := t.GetSchemaVersion()
+	if err != nil {
+		return err
+	}
+	is := d.infoCache.GetLatest()
+	if is.SchemaMetaVersion() == currVer {
```

Comment:
it come to me that when the version is latest, use the dependency maybe a quick choice, otherwise fall back to meta traveling.

Reply:
If it's in a DDL, the schema must be the latest. Why do you check it?

Reply:
campaign loop is independent thread, possibility that owner's schema is out of date exists

Reply:
Because I found a similar code in other ddl operation, i.e : https://github.com/pingcap/tidb/blob/35ef52c77cd3bec2a24027c53d01a3cf4bc2b621/ddl/table.go#L1032-L1044

Maybe it's better to follow it even if we can keep the cached inforschema is the latest now. One day if the ddl operations runs in distributed workers, it is still safe.

Reply:
> Maybe it's better to follow it even if we can keep the cached inforschema is the latest now. One day if the ddl operations runs in distributed workers, it is still safe.

You are correct.